### ISSUE TITLE
Some cmdlets return List objects which does not pass through the pipeline 'cleanly'

### ISSUE THREAD

**Issue by lfshr** [Azure/azure-powershell#3719]
### Cmdlet(s)
Get-AzureRmDisk
Get-AzureRmStorageAccountKey

### PowerShell Version
PSVersion                      5.1.14393.953
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.953
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

### Module Version
AzureRM.Automation          2.7.0
AzureRM.Batch               2.7.0
AzureRM.Compute             2.8.0
AzureRM.DataFactories       2.7.0
AzureRM.DataLakeAnalytics   2.7.0
AzureRM.Insights            2.7.0
AzureRM.Network             3.6.0
AzureRM.OperationalInsights 2.7.0
AzureRM.Profile             2.7.0
AzureRM.Resources           3.7.0
AzureRM.Tags                2.7.0

### OS Version
10.0.14393.953

### Description
Some cmdlets return a List of objects which doesn't pass through the pipeline without wrapping the command in parenthesis. I believe this list should be enumerated, and the underlying objects should be written to the pipeline instead.

### Script/Steps for Reproduction
```
Get-AzureRmDisk | Where-Object Name -eq 'GenericDiskName'
```
```
Get-AzureRmStorageAccountKey -ResourceGroupName 'GenericResourceGroupName' -Name 'GenericName' | Select-Object -ExpandProperty Key2
```

**Comment by shahabhijeet** [comment]
@lfshr I believe you were wondering about Get-AzureDisk and not Get-AzureRmDisk (typo maybe?)

**Comment by lfshr** [comment]
@shahabhijeet this refers to AzureRm.Storage\Get-AzureRmDisk. Not the classic portal equivalent. 

https://docs.microsoft.com/en-gb/powershell/module/azurerm.compute/get-azurermdisk?view=azurermps-3.7.0

**Comment by lfshr** [comment]
Beg your pardon. After searching, it's in Compute, not Storage. https://github.com/Azure/azure-powershell/blob/6732a078940ffda7e2a7b47ca7699af925a5deb7/src/ResourceManager/Compute/Commands.Compute/Generated/Disk/DiskGetMethod.cs

**Comment by lfshr** [comment]
@shahabhijeet got a fix for Get-AzureRmDisk locally. I'll create a PR for it.

**Comment by cormacpayne** [comment]
@lfshr thanks for creating pull requests to fix these issues! We will make sure the teams who own these cmdlets take a look at these changes and sign-off

**Comment by shahabhijeet** [comment]
@huangpf this issue also requires similar fix for AzureRmDisk

**Comment by lfshr** [comment]
@huangpf here's a quote from @markcowl on #3741 
>@lfshr Talked offlien with the service team - this code comes from a genrator and they have agrreed to update the generaotr to produce coee that enumerates collections, like this. Closing, as we want to update the generator rather then the generated code.

@markcowl  should this issue be closed at this point or shall we leave it open until the generator is fixed?

**Comment by markcowl** [comment]
@lfshr The generator was fixed in the last release.  Can you verify that this was fixed in 4.0?  If so we can close this.

**Comment by lfshr** [comment]
Closed as this has been fixed. Thanks for your help @markcowl.
